clang/lib/Sema/SemaChecking.cpp

```diff
@@ -7320,17 +7320,20 @@ void EquatableFormatArgument::VerifyCompatible(
                          << (ModifierFor + 1) << (Other.ModifierFor + 1),
                      FmtExpr, InFunctionCall);
       S.Diag(Other.ElementLoc, diag::note_format_cmp_with) << 0 << Other.Range;
+      return false;
     }
-    return;
+    return true;
   }
 
+  bool HadError = false;
   if (Sensitivity != Other.Sensitivity) {
     // diagnose and continue
     EmitDiagnostic(S,
                    S.PDiag(diag::warn_format_cmp_sensitivity_mismatch)
                        << Sensitivity << Other.Sensitivity,
                    FmtExpr, InFunctionCall);
     S.Diag(Other.ElementLoc, diag::note_format_cmp_with) << 0 << Other.Range;
+    HadError = true;
```

Comment:
As a nit: `HadError = S.Diag(....)` would improve readability IMO, and allow skipping some curleys.